### PR TITLE
Option to enable validating addons YAML structure

### PR DESCRIPTION
**Pull request by superseb**
https://github.com/rancher/rke/issues/776

### CODE REVIEW

File: cmd/config.go [hunk 1]
```diff
@@ -83,17 +83,18 @@ func writeConfig(cluster *v3.RancherKubernetesEngineConfig, configFile string, p
...
-	reader := bufio.NewReader(os.Stdin)
+	cluster := v3.RancherKubernetesEngineConfig{}
 
 	// Generate empty configuration file
 	if ctx.Bool("empty") {
 		cluster.Nodes = make([]v3.RKEConfigNode, 1)
 		return writeConfig(&cluster, configFile, print)
 	}
 
+	// Get cluster config from user
```
[the remark this answers is not in the document]
Comment (superseb) [reply cmd/config.go]: Doesn't need to be created before the empty bool check, it's not used in it.

File: cmd/validate.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package cmd
+
+import (
+	"github.com/rancher/rke/cluster"
+	"github.com/rancher/rke/pki"
+
+	"github.com/urfave/cli"
+)
+
+func ValidateCommand() cli.Command {
```
[the remark this answers is not in the document]
Comment (galal-hussein) [reply cmd/validate.go]: I understand the problem but the fix is partial, so i think you need to expand the fix to include checking the whole config file including the user addons

**Comment by superseb** [comment]
Removed the subcommand

**Comment by superseb** [comment]
Let me add a flag to disable the check in case it blocks someone

**Comment by superseb** [comment]
@galal-hussein Can you review? Following changed:

- Added a flag to enable behavior as it will be changing from previous versions and could break setups working on v0.1.8 as the YAML marshal used before didn't really validate
- Flag is disabled by default and can be enabled by passing `--enable-validate-useraddons`.

File: cmd/up.go [hunk 3]
```diff
@@ -218,7 +222,7 @@ func clusterUpLocal(ctx *cli.Context) error {
 
 	rkeConfig.IgnoreDockerVersion = ctx.Bool("ignore-docker-version")
 
-	_, _, _, _, _, err = ClusterUp(context.Background(), rkeConfig, nil, hosts.LocalHealthcheckFactory, nil, true, "", false, false)
+	_, _, _, _, _, err = ClusterUp(context.Background(), rkeConfig, nil, hosts.LocalHealthcheckFactory, nil, true, "", false, false, false)
```
Comment (moelsayed) [inline cmd/up.go]: Why disable it for local and dind ?

File: cmd/up.go [hunk 4]
```diff
@@ -50,6 +50,10 @@ func UpCommand() cli.Command {
 			Name:  "disable-port-check",
 			Usage: "Disable port check validation between nodes",
 		},
+		cli.BoolFlag{
+			Name:  "enable-validate-useraddons",
```
Comment (moelsayed) [inline cmd/up.go]: How about `validate-useraddons` ?

**Comment by alena1108** [comment]
@moelsayed @superseb what's the resolution on this PR? I vaguely recall the decision to validate the entire yml vs addons only

**Comment by superseb** [comment]
@moelsayed I can rebase the PR and make the changes, the question remains if we will add this single option or if we want to close this and create a new issue where we discuss how we are going to deal with validation (cluster/cluster config/yamls etc..)

**Pull request closed without merging**